manticore/core/cpu/abstractcpu.py

```diff
@@ -144,16 +146,7 @@ def size(self):
     def reg(self):
         return self._reg_name(self.op.reg)
 
-    @property
-    def type(self):
-        ''' This property encapsulate the operand type. 
-            It may be one of the following:
-                register
-                memory
-                immediate
-        '''
-        raise NotImplementedError
-        
+    @abstractmethod
```

Comment:
Better not use ABC. So we do not have the overhead of checking in each invocation.